test/matmul.jl

```diff
@@ -549,6 +549,10 @@ end
         @test mul!(copy(C), A', A, true, 2) ≈ 3C
         D = Matrix(Hermitian(A * A'))
         @test mul!(copy(D), A, A', true, 3) ≈ 4D
+        if T <: Complex
+            @test mul!(copy(C), A', A, im, 2) ≈ 2C + im*C
```

Comment:
```suggestion
            @test mul!(2C, A', A, im, 2) ≈ (4 + im) * C
```
The idea is that since `A' * A == C`, we want to ensure that the destination differs from this, so that `alpha` and `beta` aren't interchanged mistakenly.